src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/SafeExceptionConverter.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// <see cref="JsonConverter"/> for converting exceptions safely.
+    /// If the exception fails to deserialize, returns an <see cref="ExceptionDeserializationException"/> instead of failing.
+    /// </summary>
+    public class SafeExceptionConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return typeof(Exception).IsAssignableFrom(objectType);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
```

Comment:
Arg validation? Throw if objectType isn't an exception?

Reply:
JSON.NET should never pass us an object that is not an `Exception` because of the `CanConvert` function above.

Reply:
I agree with @skofman1 comments above regarding the exceptions. In my opinion the exception's data store needs to be easy to be queried and blob storage is not. We also would probably like to have an common exception data storage and we currently use AI.

Reply:
@cristinamanum Regardless of that, the infrastructure already uses exceptions in blob storage. This is merely a bug fix to having exceptions in blob storage. I think changes to the infrastructure are outside of the scope of this PR.

Reply:
@scottbommarito, understood. So we're saying "out of scope for this PR". I'm fine with that.

Reply:
Still think we need to improve validation of our public APIs, but seems to be a common theme across jobs and servercommon.